plugins/SliceInfoPlugin/SliceInfo.py

```diff
@@ -112,19 +152,11 @@ def _onWriteStarted(self, output_device):
             submitted_data = urllib.parse.urlencode(submitted_data)
             binary_data = submitted_data.encode("utf-8")
 
-            # Submit data
-            kwoptions = {"data" : binary_data,
-                         "timeout" : 1
-                         }
-            if Platform.isOSX():
-                kwoptions["context"] = ssl._create_unverified_context()
-            try:
-                f = urllib.request.urlopen(self.info_url, **kwoptions)
-                Logger.log("i", "Sent anonymous slice info to %s", self.info_url)
-                f.close()
-            except Exception as e:
-                Logger.logException("e", "An exception occurred while trying to send slice information")
+            # Sending slice info non-blocking
+            reportJob = SliceInfoThread(self.info_url, binary_data)
+            self.runningJobs.append(reportJob)
```

Comment:
In general, jobs are meant to act in a fire-and-forget manner. JobQueue manages the jobs and their assignment to worker threads, after which they are run and then destroyed. There should be no real need to keep track of them unless you need the result of a job, in which case you should use its finished signal. So I do not think you need to do all this tracking here.